docs/rules/terraform_required_providers.md

```diff
@@ -0,0 +1,46 @@
+# terraform_required_providers
+
+Require that all providers have version constraints through `required_providers` or the provider `version` attribute.
+
+## Configuration
+
+```hcl
+rule "terraform_required_providers" {
+  enabled = true
+}
+```
+
+## Example
+
+```hcl
+provider "template" {
+  version = "2"
+}
```

Comment:
This example doesn't seem to report the following warning 🤔

Reply:
Good catch!